android/src/com/frostwire/android/gui/activities/SettingsActivity.java

```diff
@@ -657,12 +658,32 @@ protected void onPostExecute(Void result) {
     @Override
     public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preference preference) {
 
-        boolean r = super.onPreferenceTreeClick(preferenceScreen, preference);
+        // If the user has clicked on a preference screen, set up the screen
         if (preference instanceof PreferenceScreen) {
-            initializePreferenceScreen((PreferenceScreen) preference);
-            currentPreferenceKey = preference.getKey();
+            setUpNestedScreen((PreferenceScreen) preference);
         }
-        return r;
+
+        return false;
+    }
+
+    public void setUpNestedScreen(PreferenceScreen preferenceScreen) {
+        final Dialog dialog = preferenceScreen.getDialog();
+
+        Toolbar nestedSettingsToolbar;
+
+        LinearLayout root = (LinearLayout) dialog.findViewById(android.R.id.list).getParent();
```

Comment:
I suggest you have a private method called:

`public void initSettingsToolbar(Context context, String title, View.OnClickListener clickListener)`

I bet you can reuse such a function for both the case above, and also for the dialog based code on the nested screen.